app/models/service_retire_task.rb

```diff
@@ -14,5 +14,43 @@ def update_and_notify_parent(*args)
   end
 
   def task_finished
+    update_attributes(:status => status == 'Ok' ? 'Completed' : 'Failed')
+  end
+
+  def task_active
+    update_attributes(:status => 'Active')
+  end
+
+  def after_request_task_create
+    update_attributes(:description => get_description)
+    parent_svc = Service.find_by(:id => options[:src_id])
+    _log.info("- creating service tasks for service <#{self.class.name}:#{id}>")
+
+    create_retire_subtasks(parent_svc)
+  end
+
+  def create_retire_subtasks(parent_service)
+    tasks = []
+    parent_service.service_resources.each do |svc_rsc|
+      next unless svc_rsc.resource.present? && svc_rsc.resource.respond_to?(:retire_now)
+      new_task = (svc_rsc.resource.type.demodulize + "RetireTask").constantize.new
+
+      next if svc_rsc.resource_type == "ServiceTemplate"
+      new_task.options.merge!(
```

Comment:
Does a `new_task` have default options that you are merging into?  If not, you can `create!` rather than `new`, merge options, set relations and `save!`.